### PR TITLE
Replace YQ with helm lint in helm test

### PR DESCRIPTION
**Pull request by jfilipcz**
#### What is this PR About?
This PR updates the way we're testing validity of  helm charts.

#### How do we test this?
TESTING.md completed succesfully

cc: @redhat-cop/bats-library

### CODE REVIEW

File: src/helm.bash [hunk 1]
```diff
@@ -24,18 +25,19 @@ helm_template() {
...
 
   # NOTE: eval is used due to helm not liking empty template_opts
-  eval "helm template ${template_opts} ${chart_dir} &> ${output_file}"
+  return_message=$(eval "helm template ${template_opts} ${chart_dir} &> ${output_file}")
 
-  # Safety check to make sure nothing above silently failed
-  if [[ ! -s "${output_file}" ]] || [[ $(wc -c "${output_file}" | awk '{print $1}') -le 1 ]] ; then
-    fail "# FATAL-ERROR: (helm.bash): File is empty: '${output_file}'" || return $?
+  # Safety check 
+  if [[ $? -eq 1 ]] ; then
```
Comment (pabrahamsson) [inline src/helm.bash:31]: I'm thinking you probably want to change this to `[[ $? -ne 0 ]]` or `[[ $? -gt 0 ]]` in order to catch all possible failures from `helm template` (unless you know that `1` is the only error code).

File: src/helm.bash [hunk 2]
```diff
@@ -24,18 +25,19 @@ helm_template() {
   output_file="${tmp_write_dir}/templates.yaml"
 
   # NOTE: eval is used due to helm not liking empty template_opts
-  eval "helm template ${template_opts} ${chart_dir} &> ${output_file}"
+  return_message=$(eval "helm template ${template_opts} ${chart_dir} &> ${output_file}")
 
-  # Safety check to make sure nothing above silently failed
-  if [[ ! -s "${output_file}" ]] || [[ $(wc -c "${output_file}" | awk '{print $1}') -le 1 ]] ; then
-    fail "# FATAL-ERROR: (helm.bash): File is empty: '${output_file}'" || return $?
+  # Safety check
```
Comment (pabrahamsson) [inline src/helm.bash:30]: Can we keep the original comment to explain what the safety check is for?

File: src/helm.bash [hunk 3]
```diff
@@ -24,18 +25,19 @@ helm_template() {
   output_file="${tmp_write_dir}/templates.yaml"
 
   # NOTE: eval is used due to helm not liking empty template_opts
-  eval "helm template ${template_opts} ${chart_dir} &> ${output_file}"
+  return_message=$(eval "helm template ${template_opts} ${chart_dir} &> ${output_file}")
 
-  # Safety check to make sure nothing above silently failed
-  if [[ ! -s "${output_file}" ]] || [[ $(wc -c "${output_file}" | awk '{print $1}') -le 1 ]] ; then
-    fail "# FATAL-ERROR: (helm.bash): File is empty: '${output_file}'" || return $?
```
Comment (pabrahamsson) [inline src/helm.bash:31]: Is it no longer necessary to check that `helm template` actually produces output?
If not, then I'd suggest getting rid of the `output_file` in this function all together as I don't see it being used for anything.

**Review by pabrahamsson — CHANGES_REQUESTED** [review]
Thanks for the update @jfilipcz, I've added a few comments/questions inline.

**Comment by jfilipcz** [comment]
> can you add a test to check the `return_message` works as expected? see:
> 
> * https://github.com/redhat-cop/bats-library/blob/master/test/tests.sh#L90

You mean to persist the `return_message` and add another check for that on top of what's already there?

File: src/helm.bash (hunk 3 above)
Comment (jfilipcz) [reply src/helm.bash:31]: `output_file` is still used for the helm.bash test itself, better to keep that

Comment (pabrahamsson) [reply src/helm.bash:31]: ah yes, I see what you're saying. All good by me.

**Review by pabrahamsson — APPROVED** [review]
Looks good, my comments have been addressed. Will wait for @garethahealy

**Review by garethahealy — APPROVED** [review]

**Pull request merged**